### PR TITLE
Apply V3 Design / Icon (issue/2132)

### PR DESCRIPTION
**Pull request by stephanschroeter**
### Pull Request Checklist

- [x] I have read and accept the [Contributor License Agreement](https://github.com/porscheofficial/oss-docs/blob/67c5b24838a293ce7a964884e6005eb71f2b8579/CONTRIBUTOR_LICENSE_AGREEMENT.md)

#### References

- Preview: 
  - https://designsystem.porsche.com/issue/2132/components/icon
  - https://designsystem.porsche.com/issue/2132/components/button-pure
  - https://designsystem.porsche.com/issue/2132/components/link-pure

#### Scope

Short description of the pull request scope

### CODE REVIEW

File: packages/components/src/components/icon/icon-styles.ts [hunk 1]
```diff
@@ -1,45 +1,118 @@
...
+  'x-small': fontSizeTextXSmall,
+  small: fontSizeTextSmall,
+  medium: fontSizeTextMedium,
+  large: fontSizeTextLarge,
+  'x-large': fontSizeTextXLarge,
 };
 
-export const getComponentCss = (color: TextColor, size: IconSize, theme: Theme): string => {
-  const dimension = sizeMap[size];
+// TODO: shall icon component support 'disabled' color too? For e.g. button-pure disabled state?
```
Comment (stephanschroeter) [inline packages/components/src/components/icon/icon-styles.ts:23]: add TODO, use tupel to reduce size

File: packages/components/src/components/icon/icon-styles.ts [hunk 2]
```diff
@@ -1,45 +1,118 @@
...
-        filter: ['dark', 'dark-electric'].includes(theme) ? 'invert(100%)' : 'none',
-      },
-      svg: {
-        fill: 'currentColor',
-        // TODO: This is a temporary fallback for Chromium and should be removed if this bug is resolved: https://bugs.chromium.org/p/chromium/issues/detail?id=1242706
-        // further information: https://melanie-richards.com/blog/currentcolor-svg-hcm/
-        '@media (forced-colors: active)': {
-          fill: 'canvasText',
-        },
+        display: 'block', // without img tag gets some extra spacing
```
Comment (stephanschroeter) [inline packages/components/src/components/icon/icon-styles.ts:99]:
```suggestion
        display: 'block', // without display, img tag gets some extra spacing
```

File: packages/components/src/components/icon/icon.examples.md [hunk 3]
```diff
@@ -52,18 +59,33 @@ can be set to `inherit` in order to specify the size from outside.
 Predefined colors associated with its theme are available. Furthermore, inherit mode can be used to define a custom
 color.
 
+<p-inline-notification heading="Important note" state="success" persistent="true">
+  The inherit mode works slightly different in v3 compared to the previous major release. A CSS filter is required to
```
Comment (stephanschroeter) [inline packages/components/src/components/icon/icon.examples.md:63]: give hint for calculator

File: packages/components/src/components/icon/icon.tsx [hunk 4]
```diff
@@ -1,24 +1,24 @@
...
+import { buildIconUrl, ICON_ARIA_ATTRIBUTES } from './icon-utils';
 import {
   AllowedTypes,
   attachComponentCss,
-  getShadowRootHTMLElement,
-  hasWindow,
-  ICON_SIZES,
+  parseAndGetAriaAttributes,
   TEXT_COLORS,
+  TEXT_SIZES,
```
Comment (stephanschroeter) [inline packages/components/src/components/icon/icon.tsx:9]: Use alias

File: packages/components/src/components/icon/icon.tsx [hunk 5]
```diff
@@ -1,24 +1,24 @@
...
   validateProps,
 } from '../../utils';
-import type { IconName, IconSize, PropTypes, SelectedAriaAttributes, TextColor, Theme } from '../../types';
+import type { IconName, PropTypes, SelectedAriaAttributes, TextColor, TextSize, Theme } from '../../types';
 import { getComponentCss } from './icon-styles';
+import { ICON_NAMES } from '@porsche-design-system/icons';
 
 const propTypes: PropTypes<typeof Icon> = {
-  name: AllowedTypes.string, // TODO: should use object keys of ICON_MANIFEST as soon as keys are available with param-case syntax
+  name: AllowedTypes.oneOf<IconName>(ICON_NAMES),
```
Comment (stephanschroeter) [inline packages/components/src/components/icon/icon.tsx:18]: use object keys

**Review by denyo — APPROVED** [review]

File: packages/components/src/index.html [hunk 6]
```diff
@@ -15,7 +15,269 @@
       <h1>Porsche Design System - Playground</h1>
     </header>
 
-    <main></main>
+    <main>
```
Comment (stephanschroeter) [inline packages/components/src/index.html:18]: revert

**Pull request merged**